spec/furniture/marketplace/product_tags_system_spec.rb

```diff
@@ -0,0 +1,49 @@
+require "rails_helper"
+
+describe "Product Tags", type: :system do
+  let(:space) { create(:space, :with_entrance, :with_members) }
+  let(:marketplace) { create(:marketplace, :ready_for_shopping, room: space.entrance) }
```

Comment:
These two lines are repeated in pretty much every `marketplace/*_system_spec.rb`; maybe we pull them to a mixin?

Reply:
I don't have a strong opinion either way.

For tests like these, I don't mind duplicating a couple of lines, and I like seeing at a glance how data is being setup for the test. An intermediate solution would be to have a trait like `:with_full_space` on the Marketplace factory that creates a Marketplace with a fully-setup and populated space.

Reply:
I like the solution of a `marketplace` trait for making instantiating a `marketplace` in a space with an entrance + members.